### ISSUE TITLE
UnnecessaryInnerClass false positive

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#9072]
## Expected Behavior
No flag.

## Observed Behavior
Flagged and recommendation shows error in IDE and breaks tests (they don't run).

## Steps to Reproduce
```kotlin
class SomeFileKtUnitTest {
	@Nested
	inner class Group1 {
            @Test test1() {}
            @Test test2() {}
        }
	@Nested
	inner class Group2 {
            @Test test3() {}
            @Test test4() {}
        }
}
```

## Context
https://docs.junit.org/5.4.1/api/org/junit/jupiter/api/Nested.html

## Your Environment
* Version of detekt used: 2.0.0-alpha02
* Version of Gradle used (if applicable): 9.3
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: any
* Link to your project (if it's a public repository):

**Comment by BraisGabin** [comment]
This is junit specific, there's nothing we can do from detekt. I recommend you to use `ignoreAnnotated` with `Nested`.

**Comment by TWiStErRob** [comment]
Ah, good point, I forgot about suppressors, it didn't list the option in UnnecessaryInnerClass docs.
